File: pinyougou_parent/pinyougou_sellergoods_interface/src/main/java/com/pinyougou/sellergoods/service/CitiesService.java
package com.pinyougou.sellergoods.service;
import java.util.List;
import com.pinyougou.pojo.TbCities;							  
import entity.PageResult;
import entity.Result;
/**
 * cities服务层接口
 * @author jieweili
 *
 */
public interface CitiesService {

    /**
     * 返回全部列表(已弃置)
     * @return
     */
    List<TbCities> findAll();
 
    /**
     * 返回分页对象(过时)
     * @param pageNum
     * @param pageSize
     * @return
     */
	 
    PageResult findPage(Integer pageNum, Integer pageSize);

    /**
     * 添加数据
     * @param cities
     * @return
     */
    public Result saveCities(TbCities cities);

    /**
     * 根据id查询一个对象
     * @param id
     * @return
     */
    TbCities getCitiesById(Integer id);

    /**
     * 修改对象
     * @param cities
     * @return
     */
    Result updateCities(TbCities cities);

    /**
     * 批量删除
     * @param idList
     * @return
     */
    Result removeCities(Integer[] idList);

    /**
     * 根据条件分页查询
     * @param cities
     * @param pageNum
     * @param pageSize
     * @return
     */
    PageResult listCitiesByExample(TbCities cities, Integer pageNum, Integer pageSize);
 
}

File: pinyougou_parent/pinyougou_manager_web/src/main/webapp/js/service/citiesService.js
app.service('citiesService', function ($http) {
    //旧的查询方法(弃置)								   
    this.findPage = function (pageNum, pageSize) {
        return $http.get('../cities/findPage.do?pageNum=' + pageNum + '&pageSize=' + pageSize)
    };

    //查询方法,读取列表数据绑定到表单中
    this.search = function (pageNum, pageSize, searchEntity) {
        return $http.post('../cities/findPage.do?pageNum=' + pageNum + "&pageSize=" + pageSize, searchEntity)
    };

    //删除方法
    this.removeCitiesList = function (idList) {
        return $http.get('../cities/removeCities.do?idList=' + idList)
    };

    //根据id查询
    this.findOne = function (id) {
        return $http.get('../cities/findOne.do?id=' + id)
    };

    //修改品牌
    this.updateCities = function (cities) {
        return $http.post('../cities/updateCities.do', cities)
    };

    //新增品牌
    this.saveCities = function (cities) {
        return $http.post('../cities/saveCities.do', cities)
    };   
});
